hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/SecureBulkLoadManager.java

```diff
@@ -382,22 +382,27 @@ public String prepareBulkLoad(final byte[] family, final String srcPath, boolean
         throw new IOException("Path does not reference a file: " + p);
       }
 
-      // Check to see if the source and target filesystems are the same
-      if (!FSUtils.isSameHdfs(conf, srcFs, fs)) {
-        LOG.debug("Bulk-load file " + srcPath + " is on different filesystem than " +
+      try {
```

Comment:
The whole code with HDFS should be all considered as IOException when error happened, instead of just throwing IOException when rename failed;

Reply:
When we throw IOE from here, it wont get retried from client end?  Did not check code in detail.

Reply:
[ BulkLoadHFilesTool.java#396](https://github.com/apache/hbase/blob/a9205f8f4d98ee672c0c6aa9cafa5ef2afc6aab5/hbase-server/src/main/java/org/apache/hadoop/hbase/tool/BulkLoadHFilesTool.java#L397)
This is the cliend-side retry logic if exception/error happened.

Reply:
From the code , we could find that if exception is thrown correctly instead of swallowed, the retry attempts number could be controlled by `hbase.client.retries.number` and `hbase.bulkload.retries.retryOnIOException` instead of falling into a retry disaster and user cannot do anything because the retry attempts number are controlled by number of regions.

Reply:
Overall, in this part, I just extend the exception catch domain from previous just `fs.rename()` to the whole HDFS operation, they need obviously the same process logic because they are all hdfs operations;